src/config.rs

```diff
@@ -0,0 +1,69 @@
+use std::{borrow::Cow, path::PathBuf};
+
+use figment::{
+    providers::{Env, Format, Yaml},
+    Figment,
+};
+
+use serde::{Deserialize, Serialize};
+
+#[derive(PartialEq, Debug, Serialize, Deserialize)]
+pub struct Config {
+    /// The sentry DSN to use for error reporting.
+    pub sentry_dsn: Option<String>,
+
+    /// The environment to report to sentry errors to.
+    pub sentry_env: Option<Cow<'static, str>>,
+}
+
+impl Config {
+    /// Load configuration from an optional configuration file and environment
+    pub fn extract(path: &Option<PathBuf>) -> anyhow::Result<Config> {
+        let builder = Figment::new();
+
+        let builder = match path {
+            Some(path) => builder.merge(Yaml::file(path)),
+            _ => builder,
+        };
```

Comment:
```suggestion
        let mut builder = Figment::new();

        if let Some(path) = path {
            builder = builder.merge(Yaml::file(path));
        }
```

Reply:
Yeah i had originally done thus but didn't like that it was mutable